external/anomaly/ote_anomalib/openvino.py

```diff
@@ -172,16 +159,38 @@ def infer(self, dataset: DatasetEntity, inference_parameters: InferenceParameter
             anomaly_map, pred_score = self.inferencer.predict(
                 dataset_item.numpy, superimpose=False, meta_data=meta_data
             )
+            # TODO: inferencer should return predicted label and mask
+            pred_label = pred_score >= 0.5
+            pred_mask = (anomaly_map >= 0.5).astype(np.uint8)
+            probability = pred_score if pred_label else 1 - pred_score
             if self.task_type == TaskType.ANOMALY_CLASSIFICATION:
-                annotations_scene = self.annotation_converter.convert_to_annotation(pred_score, meta_data)
-            elif self.task_type in (TaskType.ANOMALY_DETECTION, TaskType.ANOMALY_SEGMENTATION):
-                annotations_scene = self.annotation_converter.convert_to_annotation(anomaly_map, meta_data)
+                label = self.anomalous_label if pred_score >= 0.5 else self.normal_label
+                dataset_item.append_labels([ScoredLabel(label=label, probability=float(probability))])
+            elif self.task_type == TaskType.ANOMALY_SEGMENTATION:
+                annotations = create_annotation_from_segmentation_map(
+                    pred_mask, anomaly_map.squeeze(), {0: self.normal_label, 1: self.anomalous_label}
+                )
+                dataset_item.append_annotations(annotations)
+                if len(annotations) == 0:
+                    dataset_item.append_labels([ScoredLabel(label=self.normal_label, probability=float(probability))])
+                else:
+                    dataset_item.append_labels(
+                        [ScoredLabel(label=self.anomalous_label, probability=float(probability))]
+                    )
```

Comment:
```suggestion
                label = self.normal_label if len(annotations) == 0 else self.anomalous_label 
                dataset_item.append_labels([ScoredLabel(label=label, probability=float(probability))])
```
and `dataset_item.append_labels` can be moved to the outside of the if-else clause, because it's the same for all branches.

Reply:
Good catch. I've made the changes